src/main/java/org/opentripplanner/routing/core/RoutingContext.java

```diff
@@ -384,6 +384,13 @@ private void setServiceDays() {
                 addIfNotExists(this.serviceDays, new ServiceDay(graph, serviceDate, calendarService, agency.getId()));
                 addIfNotExists(this.serviceDays, new ServiceDay(graph, serviceDate.next(),
                         calendarService, agency.getId()));
+                if (opt.serviceDayLookout > 0) {
+                    ServiceDate sd = serviceDate;
+                    for (int i = 0; i < opt.serviceDayLookout; i++) {
+                        sd = opt.arriveBy ? sd.previous() : sd.next();
+                        addIfNotExists(this.serviceDays, new ServiceDay(graph, sd, calendarService, agency.getId()));
```

Comment:
This adds only services on one side of the specified search date (after or before, but not both). So this handles cases where the service occurs far in the future or past, but not cases where the service day is multiple calendar days long (e.g. ferry running for multiple days as in Norway). This is probably an intentional choice, but these expected use cases (and cases that are not covered) should be explained in the documentation or Javadoc on the new parameters.

Reply:
Hm. Would it address the multiple-day issue if we made the serviceDayLookout symmetric? The tradeoff is that iterating through ServiceDays in `TransitBoardAlight` is a hot spot. We could also add "forwardLookout" and "backwardLookout" parameters, but that would increase the complexity a bit. I guess I lean toward adding both parameters with sane defaults (1) and letting power users change them. Maybe this is something that should just be configurable via `router-config.json` and not via the API resource. Thoughts?

Reply:
I don't even think we necessarily need to handle the very long multi-day trips here. It's just something that came up in Norway that also involves increasing the number of service days, so I think we need to state clearly in docs what problem is being solved here. If it's only boarding trips many days in the future (but not boarding trips many days long) that's fine, I just think it will be helpful to future readers to make that clear.

If you do want to add a backwardLookout, I'd discuss with @t2gran or @gmellemstrand to see if this would work in their use case. 

It is debatable whether this should be changed in router-config or per-request in the API. The main problem I see with doing it in the API is that you can grind the server to a halt with this parameter, so it's a kind of DDOS target. The problem I see with doing it in router-config is that some special trips really need to be multi-day, and most others do not. So it could lead to a ton of extra calculation where not necessary.